src/zimdump.cpp

```diff
@@ -291,6 +291,18 @@ void ZimDumper::writeHttpRedirect(const std::string& directory, const std::strin
     write_to_file(directory + SEPARATOR, outputPath, content.c_str(), content.size());
 }
 
+static bool testSymlink(const std::string& sym, const std::string& target) {
+  char buf[256];
```

Comment:
Why 256 ?
Symlink can be longer than 256, we should handle that.

Has we are C++17, we may simply use `std::filesystem::read_link` (https://en.cppreference.com/w/cpp/filesystem/read_symlink) (which may involve a more substantial change in our code base)

Reply:
> <b>readlink</b>() does not append a terminating null byte to <i>buf</i>.  It will (silently) truncate the contents (to a length of <i>bufsiz</i> characters), in case the buffer is too small to hold all of the contents.

I read man page of [readlink](https://man7.org/linux/man-pages/man2/readlink.2.html), so I want buffer to have one more byte than default 255(assuming hardcoded MAX_PATH) so that return value of <b>readlink</b>() can be **safely** placing trailing NULL character in buf.

I agree to use <code class="notranslate">std::filesystem::read_link</code> which solves truncation of longer path of <b>readlink</b> and also conditional compilation of Windows, I guess(Sorry, no win32 compiling experience for years.)

Reply:
> I read man page of [readlink](https://man7.org/linux/man-pages/man2/readlink.2.html), so I want buffer to have one more byte than default 255(assuming hardcoded MAX_PATH) so that return value of readlink() can be safely placing trailing NULL character in buf.

Yes, but assuming MAX_PATH is 256 is wrong. On my linux it is 4096. 256 is most of the time the max filename (so the size of one component in the path).
And even assuming 4096 is wrong, see https://insanecoding.blogspot.com/2007/11/pathmax-simply-isnt.html

The solution is to have a increasing buffer (`vector<char>` is a nice canditate) and increase the buffer until returned value of `readlink()` is smaller than the bufsize.
(Or let `std::filesystem::read_link` do this for us)

Reply:
@mgautierfr  Absolutely! I have created [another PR](https://github.com/openzim/zim-tools/pull/417#issue-2366920215) to avoid these issues. Please kindly take a look. Thanks.